src/Microsoft.AspNet.Mvc.Core/Formatters/OutputFormatter.cs

```diff
@@ -256,17 +284,18 @@ public virtual void WriteResponseHeaders(OutputFormatterWriteContext context)
         protected string GetMediaTypeWithCharset(string mediaType, Encoding encoding)
         {
             var mediaTypeEncoding = MediaTypeEncoding.GetEncoding(mediaType);
-            if (mediaTypeEncoding == encoding)
+            if (mediaTypeEncoding != null &&
+                mediaTypeEncoding.WebName.Equals(encoding.WebName, StringComparison.OrdinalIgnoreCase))
```

Comment:
I changed the test for the cache to avoid using mocks, this is covered now by that test